services/telemetry/telemetry.go

```diff
@@ -64,33 +72,77 @@ const (
 	TRACK_PLUGINS  = "plugins"
 )
 
-// declaring this as var to allow overriding in tests
-var SENTRY_DSN = "placeholder_sentry_dsn"
+type ServerIface interface {
+	Config() *model.Config
+	IsLeader() bool
+	GetPluginsEnvironment() *plugin.Environment
+	License() *model.License
+	GetRoleByName(string) (*model.Role, *model.AppError)
+	GetSchemes(string, int, int) ([]*model.Scheme, *model.AppError)
+}
 
-func (s *Server) SendDailyDiagnostics() {
-	s.sendDailyDiagnostics(false)
+type TelemetryService struct {
+	srv                        ServerIface
+	dbStore                    store.Store
+	searchEngine               *searchengine.Broker
+	log                        *mlog.Logger
+	rudderClient               rudder.Client
+	TelemetryID                string
+	timestampLastTelemetrySent time.Time
 }
 
-func (s *Server) sendDailyDiagnostics(override bool) {
-	if *s.Config().LogSettings.EnableDiagnostics && s.IsLeader() && ((!strings.Contains(RUDDER_KEY, "placeholder") && !strings.Contains(RUDDER_DATAPLANE_URL, "placeholder")) || override) {
-		s.initDiagnostics(RUDDER_DATAPLANE_URL)
-		s.trackActivity()
-		s.trackConfig()
-		s.trackLicense()
-		s.trackPlugins()
-		s.trackServer()
-		s.trackPermissions()
-		s.trackElasticsearch()
-		s.trackGroups()
-		s.trackChannelModeration()
+func New(srv ServerIface, dbStore store.Store, searchEngine *searchengine.Broker, log *mlog.Logger) *TelemetryService {
+	service := &TelemetryService{
+		srv:          srv,
+		dbStore:      dbStore,
+		searchEngine: searchEngine,
+		log:          log,
 	}
+	service.ensureTelemetryID()
+	service.initRudder(RUDDER_DATAPLANE_URL)
+	return service
 }
 
-func (s *Server) SendDiagnostic(event string, properties map[string]interface{}) {
-	if s.rudderClient != nil {
-		s.rudderClient.Enqueue(rudder.Track{
+func (ds *TelemetryService) ensureTelemetryID() {
```

Comment:
Not an important point at all, but why `ds`?

Reply:
Good question, it was `DiagnosticsService` before, not makes a lot of sense to change everything to `ts`.